redaxo/src/core/lib/util/markdown.php

```diff
@@ -38,6 +29,107 @@ public static function factory()
      */
     public function parse($code)
     {
-        return $this->parser->text($code);
+        $parser = new ParsedownExtra();
+        $parser->setBreaksEnabled(true);
+
+        return $parser->text($code);
+    }
+
+    /**
+     * Parses markdown code and extracts a table-of-content.
+     *
+     * @param string $code Markdown code
+     *
+     * @return array tupel of table-of-content and content
+     */
+    public function parseWithToc($code)
+    {
+        $parser = new rex_parsedown();
+        $parser->setBreaksEnabled(true);
+
+        $content = $parser->text($code);
+        $headers = $parser->headers;
+
+        $previous = 1;
+        $toc = '';
+
+        foreach ($headers as $header) {
+            $level = $header['level'];
+
+            if ($level > $previous) {
+                $toc .= "<ul>\n";
+                $previous = $level;
+            } elseif ($level < $previous) {
+                $toc .= "</li>\n";
+                $toc .= "</ul>\n";
+                $previous = $level;
+            } else {
+                $toc .= "</li>\n";
+            }
+
+            $toc .= "<li>\n";
+            $toc .= '<a href="#'.$header['id'].'">'.$header['text']."</a>\n";
+        }
+
+        for (; $previous > 1; --$previous) {
+            $toc .= "</li>\n";
+            $toc .= "</ul>\n";
+        }
+
+        return [$toc, $content];
+    }
+}
+
+/**
+ * @internal
+ */
+class rex_parsedown extends ParsedownExtra
+{
+    private $ids = [];
+
+    public $headers = [];
+
+    protected function blockHeader($line)
+    {
+        $block = parent::blockHeader($line);
+
+        return $this->handleHeader($block);
+    }
+
+    protected function blockSetextHeader($line, array $block = null)
+    {
+        $block = parent::blockSetextHeader($line, $block);
+
+        return $this->handleHeader($block);
+    }
+
+    private function handleHeader(array $block)
+    {
```

Comment:
Hier bekomme ich einen Fehler, da `$block` null sein kann. Wird in Zeile 103 übergeben.

Reply:
Hast du ein Beispiel, wann das passiert?

Reply:
Mit weiter unten stehendem Beispiel